packages/toolkit/src/entities/unsorted_state_adapter.ts

```diff
@@ -158,7 +158,9 @@ export function createUnsortedStateAdapter<T>(
         0
 
       if (didMutateIds) {
-        state.ids = state.ids.map((id) => newKeys[id] || id)
+        state.ids = Array.from(
```

Comment:
The stat.ids iteration is necessary? What do you think about:
`state.ids = Object.keys(state.entities)` ?

Reply:
I _think_ that could work, actually. Worth a shot?

Reply:
Updated and all the tests are passing too